config/r4e/v1_0/translate.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.)
+
+package v1_0
+
+import (
+	"github.com/coreos/butane/config/common"
+	cutil "github.com/coreos/butane/config/util"
+	"github.com/coreos/butane/translate"
+
+	"github.com/coreos/ignition/v2/config/v3_3/types"
+	"github.com/coreos/vcontext/path"
+	"github.com/coreos/vcontext/report"
+)
+
+// ToIgn3_3Unvalidated translates the config to an Ignition config.  It also
+// returns the set of translations it did so paths in the resultant config
+// can be tracked back to their source in the source config.  No config
+// validation is performed on input or output.
+func (c Config) ToIgn3_3Unvalidated(options common.TranslateOptions) (types.Config, translate.TranslationSet, report.Report) {
+	ret, ts, r := c.Config.ToIgn3_3Unvalidated(options)
+	if r.IsFatal() {
+		return types.Config{}, translate.TranslationSet{}, r
+	}
+
+	checkForForbiddenFields(ret, &r)
+
+	return ret, ts, r
+}
+
+// Checks and adds the appropiate errors when unsupported fields on r4e are
+// provided
+func checkForForbiddenFields(t types.Config, r *report.Report) {
+	for i := range t.KernelArguments.ShouldExist {
+		r.AddOnError(path.New("path", "json", "kernel_arguments", "should_exist", i), common.ErrGeneralKernelArgumentSupport)
+	}
+	for i := range t.KernelArguments.ShouldNotExist {
+		r.AddOnError(path.New("path", "json", "kernel_arguments", "should_not_exist", i), common.ErrGeneralKernelArgumentSupport)
+	}
+	for i := range t.Storage.Directories {
```

Comment:
It seems surprising that files are supported but directories and links aren't, since they're all in the class of "filesystem objects".  For example, Ignition will implicitly create parent directories for files, but without explicit `Directories` support there won't be a way of specifying their ownership or permissions.  What's the reasoning here?

Reply:
let me ping @runcom and @miabbott so that we can see what we do here since that's a good point.

Reply:
yep, we want those, my mistake, will update in a moment :+1: